src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala

```diff
@@ -48,17 +50,24 @@ class AppsResource @Inject() (
   import Normalization._
 
   private[this] val ListApps = """^((?:.+/)|)\*$""".r
-  private implicit lazy val appDefinitionValidator = AppDefinition.validAppDefinition(config.availableFeatures)(pluginManager)
 
-  private val normalizationConfig = AppNormalization.Configuration(
-    config.defaultNetworkName.toOption,
-    config.mesosBridgeName())
+  private def createValidatorAndNormalizerForApp(pathId: PathId): (Normalization[raml.App], Validator[AppDefinition]) = {
+    val roleSettings = RoleSettings.forService(config, pathId, groupManager.rootGroup())
+    val normalizationConfig = AppNormalization.Configuration(config, roleSettings)
+    implicit val normalizer: Normalization[raml.App] = appNormalization(normalizationConfig)(AppNormalization.withCanonizedIds())
+    implicit val validator: Validator[AppDefinition] = AppDefinition.validAppDefinition(config.availableFeatures, roleSettings)(pluginManager)
```

Comment:
There is no need to make it `implicit` since you are returning them explicitly anyways.

Reply:
True, was left over from C&P. Fixed